server/src/services/soccer.service.ts

```diff
@@ -116,4 +120,38 @@ export const soccerService = {
 
         return validatedData.response;
     },
+
+    transformFieldsToActualGenerationParams: async (
+        fieldsToTransform: Pick<PackagesGenerationParamsFromFreeText, 'teams' | 'league'>
+    ) => {
+        const { league, teams } = fieldsToTransform;
+        let actualLeague: League | undefined;
+        let actualTeams: Team[] | undefined;
+
+        if (league) {
+            const leagues = await soccerService.getLeaguesByName(league);
```

Comment:
```suggestion
            const leaguesParams = await soccerService.getLeaguesByName(league);
```
call it `leaguesParams` or `actualLeagues` just to make it diff from league so it wont be confusing